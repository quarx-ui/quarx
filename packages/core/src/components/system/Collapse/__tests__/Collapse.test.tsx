import { render, screen } from '@testing-library/react';
import { Collapse } from '@core';

describe('Collapse', () => {
    it('text should be in the document', () => {
        render(<Collapse open>traffic jams</Collapse>);

        expect(screen.queryByText('traffic jams')).toBeInTheDocument();
    });

    it('default layout', () => {
        const { asFragment } = render(
            <Collapse open={false}>
                Traffic jams
            </Collapse>,
        );

        expect(asFragment()).toMatchSnapshot();
    });
});

describe('Collapse opened', () => {
    it('text should be in the document', () => {
        render(<Collapse open>traffic jams</Collapse>);

        expect(screen.queryByText('traffic jams')).toBeInTheDocument();
    });
});

describe('Collapse closed 10px vertical', () => {
    it('text should be in the document', () => {
        render(<Collapse open={false} collapsedSize={10}>traffic jams</Collapse>);

        expect(screen.queryByText('traffic jams')).toBeInTheDocument();
    });
});

describe('Collapse closed 10px horizontal', () => {
    it('text should be in the document', () => {
        render(
            <Collapse open={false} orientation="horizontal" collapsedSize={10}>
                traffic jams
            </Collapse>,
        );

        expect(screen.queryByText('traffic jams')).toBeInTheDocument();
    });
});
