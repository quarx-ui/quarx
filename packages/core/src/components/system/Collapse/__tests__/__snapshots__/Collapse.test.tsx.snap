// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Collapse default layout 1`] = `
<DocumentFragment>
  <div
    class="QxCollapse css-1w91dc0-QxCollapse-root-QxCollapse-root"
  >
    <div
      class="QxCollapse-wrapper css-43rt2t-QxCollapse-wrapper"
    >
      Traffic jams
    </div>
  </div>
</DocumentFragment>
`;
